spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-local/src/main/java/org/springframework/cloud/data/module/deployer/local/LocalModuleDeployer.java

```diff
@@ -77,9 +79,12 @@ public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
 		catch (Exception e) {
 			throw new IllegalStateException("failed to determine URL for module: " + module, e);
 		}
-		args.add(String.format("--%s.server.port=%d", module, port));
-		args.add(String.format("--%s.endpoints.shutdown.enabled=true", module));
-		launcher.launch(new String[] { module }, args.toArray(new String[args.size()]));
+		args.put("server.port", String.valueOf(port));
+		args.put("endpoints.shutdown.enabled", "true");
+
+		ModuleLaunchRequest moduleLaunchRequest = new ModuleLaunchRequest(module);
+		moduleLaunchRequest.setArguments(args);
```

Comment:
I think we should make the `ModuleLaunchRequest` accept the arguments as a 2nd constructor arg. We can remove the setter, but keep the *add* method. That would make it less likely to accidentally forget passing args as you did here at first.